source/MaterialXContrib/MaterialXNode/Plugin.cpp

```diff
@@ -31,8 +33,22 @@ void Plugin::initialize(const std::string& loadPath)
     _resourceSearchPath.append(searchPath / MaterialX::FilePath("../../resources"));
     _resourceSearchPath.append(searchPath / MaterialX::FilePath("../resources"));
 
-    // Set to resource path for now
-    _shaderDebugPath = _resourceSearchPath[1];
+    MHWRender::MRenderer* const theRenderer = MHWRender::MRenderer::theRenderer();
+    MHWRender::MFragmentManager* const fragmentManager = theRenderer ? theRenderer->getFragmentManager() : nullptr;
+    if (!fragmentManager)
+    {
+        MGlobal::displayError("Failed to get the VP2 fragment manager");
+    }
+    else
+    {
+        // Set to resource path for now
+        std::string shaderDebugPath = _resourceSearchPath[1].asString();
+
+        // Add explicitly as VP2 does no prepend a folder separator, and thus fails silently to output anything.
+        shaderDebugPath += "/";
+        fragmentManager->setEffectOutputDirectory(shaderDebugPath.c_str());
```

Comment:
Since we need the change to these paths to persist, perform it once on plugin initialization. Previously, this happened every time a fragment was registered. Plus, the shader debug path no longer needs to be exposed.